camel/storages/graph_storages/neo4j_graph.py

```diff
@@ -339,6 +344,135 @@ def refresh_schema(self) -> None:
             ]
         )
 
+    def add_node(
+        self, label: str, properties: Dict[str, Any] | None = None
+    ) -> None:
+        r"""Adds a node to the Neo4j database.
+
+        Args:
+            label (str): The label for the node, representing
+                its type or category.
+            properties (Optional[Dict[str, Any]]): A dictionary of properties
+                to be assigned to the node.
+        """
+        if properties is None:
+            properties = {}
+
+        # Generate a unique UUID for the node if not present
+        if 'uuid' not in properties:
+            properties['uuid'] = str(uuid.uuid4())
+
+        # Construct the Cypher query to create a node
+        # with given label and properties
+        query = f"CREATE (n:{label} {{"
+        query += ", ".join([f"{key}: ${key}" for key in properties.keys()])
+        query += "}) RETURN n.uuid as node_uuid"
+
+        try:
+            with self.driver.session(database=self.database) as session:
+                result = session.run(query, properties).single()
+                if result:
+                    node_uuid = result["node_uuid"]
+                    self._add_relationship_to_previous(node_uuid)
+                    self.previous_node_id = node_uuid
+                    logger.info(
+                        f"Node with label '{label}' added successfully."
+                    )
+                    return node_uuid
+                else:
+                    logger.error(
+                        f"Failed to add node with label '{label}': "
+                        "No result found."
+                    )
+                    raise ValueError("Node creation failed: No result found.")
+        except Exception as e:
+            logger.error(f"Failed to add node with label '{label}': {e}")
+            raise
+
+    def _add_relationship_to_previous(self, current_node_uuid: str) -> None:
+        if self.previous_node_id is None:
+            return
+
+        query = (
+            "MATCH (prev {uuid: $prev_uuid}), (current {uuid: $current_uuid}) "
+            "CREATE (prev)-[:NEXT]->(current)"
+        )
+
+        try:
+            with self.driver.session(database=self.database) as session:
+                session.run(
+                    query,
+                    {
+                        "prev_uuid": self.previous_node_id,
+                        "current_uuid": current_node_uuid,
+                    },
+                )
+                logger.info(
+                    f"Created relationship from {self.previous_node_id} "
+                    f"to {current_node_uuid}."
+                )
```

Comment:
this relationship is specially designed for message sequence, but I think we use KG is to get relationship between entities of the message content, rather than just retrieve linked message back, which is no much difference with `ChatHistoryBlock`